src/modules/wallet/utils.ts

```diff
@@ -49,3 +50,21 @@ function getWallets(
 export function isLedgerWallet() {
   return eth.wallet instanceof wallets.LedgerWallet
 }
+
+export function isApprovableWallet() {
+  const { ethereum } = window as EthereumWindow
+  return ethereum !== undefined && typeof ethereum.enable === 'function'
+}
+
+export async function isWalletApproved() {
+  const { ethereum } = window as EthereumWindow
+
+  if (ethereum === undefined) {
+    return false
+  }
+
+  // `isApproved` is not standard. It's supported by MetaMask and it's expected to be implemented by other wallet vendors
+  // but we need to check just in case.
+  const aprobable = ethereum._metamask || ethereum
+  return aprobable.isApproved ? await aprobable.isApproved() : true
```

Comment:
as of latest MetaMask build, the default value for this method should be false

https://github.com/MetaMask/metamask-extension/pull/4703#issuecomment-434518465